sdk/docs/source/daml/code-snippets/Batching.daml

```diff
@@ -0,0 +1,49 @@
+-- Copyright (c) 2024 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+-- SPDX-License-Identifier: Apache-2.0
+
+module Batching where
+
+import DA.Foldable
+
+doUpdate : Bool -> Party -> Party -> [ContractId Example] -> Update ()
+
+doUpdate efficient owner obs batch = do
+   if efficient then do
+      batcher <- create BatchFoo with owner
+      -- This only works out if obs is the same observer on all the exercised Example contracts
+      batcher `exercise` Run with batch = batch; obs = obs
+   else do
+      -- Canton will create one view per exercise
+      forA_ batch (`exercise` Foo)
+   pure ()
+
+template Example
+   with
+      owner : Party
+      obs: Party
+   where
+      signatory owner
+      observer obs
+
+      choice Foo : ()
+         controller owner
+         do
+         return ()
```

Comment:
I'm surprised that indentation complies. Normally, the `do` is indented less than the body. Is this file anywhere checked for compilation by CI?

Reply:
I'd claim that it does, but not exactly sure where. However, I tried it locally with VSC and got no syntax error and compilation worked :).